Shell/Shell.cpp

```diff
@@ -1835,3 +1835,26 @@ void Shell::stop_all_jobs()
         }
     }
 }
+
+void Shell::save_to(JsonObject& object)
+{
+    Core::Object::save_to(object);
+    object.set("working_directory", cwd);
+    object.set("username", username);
+    object.set("user_home_path", home);
+    object.set("user_id", uid);
+    object.set("directory_stack_size", directory_stack.size());
+    object.set("cd_history_size", cd_history.size());
+
+    // Jobs.
+    object.set("number of jobs", jobs.size());
```

Comment:
How about we put the jobs in a `JsonArray` instead? Then we don't need a separate value for the count.